src/plugins/lua/MQ2Lua.cpp

```diff
@@ -1989,7 +1989,13 @@ PLUGIN_API void OnUpdateImGui()
 			if (IGFD_IsOk(s_scriptLaunchDialog))
 			{
 				IGFD_Selection selection = IGFD_GetSelection(s_scriptLaunchDialog, IGFD_ResultMode_KeepInputFile);
-				char* selected_file = selection.table->filePathName;
+
+				// avoid a silent crash to desktop by ensuring selection.table is valid
+				char* selected_file = (selection.table != nullptr) ? (
+					(selection.table->filePathName != nullptr) ? selection.table->filePathName : (
+						(selection.table->fileName != nullptr) ? selection.table->fileName : nullptr
+					)
+				) : nullptr;
```

Comment:
This is an overuse of ternary expression which makes it borderline unreadable. It would be much easier to read if it was just using `if` statements. Can you rewrite it?

Reply:
char* selected_file = nullptr;
if (selection.table) {
    selected_file = selection.table->filePathName ? selection.table->filePathName : selection.table->fileName;
}

Reply:
```cpp
char* selected_file = nullptr;
if (selection.table != nullptr) {
    if (selection.table->filePathName != nullptr) {
        selected_file = selection.table->filePathName;
    } else if (selection.table->fileName != nullptr) {
        selected_file = selection.table->fileName;
    }
}
```

Reply:
> ```c++
> char* selected_file = nullptr;
> if (selection.table != nullptr) {
>     if (selection.table->filePathName != nullptr) {
>         selected_file = selection.table->filePathName;
>     } else if (selection.table->fileName != nullptr) {
>         selected_file = selection.table->fileName;
>     }
> }
> ```

unnecessary if you are confident in the non-null status of fileName